### ISSUE TITLE
[symfony/translation] Error in readLocalTranslations()

### ISSUE THREAD

**Issue by tomasz-kusy** [symfony/symfony#44185]
### Symfony version(s) affected

5.3

### Description

Method readLocalTranslations has wrongly implemented `filterCatalogue` method;
In TranslationTrait 
[near line 35](https://github.com/symfony/translation/blob/17a965c8f3b1b348cf15d903ac53942984561f8a/Command/TranslationTrait.php#L35)
we have:
```
                foreach ($domains as $domain) {
                    $catalogue = $this->filterCatalogue($catalogue, $domain);
                    $bag->addCatalogue($catalogue);
                }
```
The problem is:
```
                    $catalogue = $this->filterCatalogue($catalogue, $domain);
```
If there are more than one domain, after the first run of the loop we got filtered $catalogue. So we can't find second domain.

### How to reproduce

try to to use readLocalTranslations() with more than one $domain

### Possible Solution

```
                foreach ($domains as $domain) {
                    $catalogue = $this->filterCatalogue($catalogue, $domain);
                    $bag->addCatalogue($catalogue);
                }
```
change to:
```
                foreach ($domains as $domain) {
                    $newCatalogue = $this->filterCatalogue($catalogue, $domain);
                    $bag->addCatalogue($newCatalogue);
                }
```

**Comment by tomasz-kusy** [comment]
Any news? I think that this error is very obvious :)

**Comment by fabpot** [comment]
Would you like to submit a PR?

**Comment by tomasz-kusy** [comment]
Sure

**Comment by tomasz-kusy** [comment]
@fabpot As I see now it's fixed in 5.4. I'm not sure about versions maintenance policy... Submit PR for 5.3?

**Comment by nicolas-grekas** [comment]
Yes please, 5.3 is still maintained.
